src/utils.js

```diff
@@ -121,28 +121,34 @@ const ipRangeBlacklist = [
 ];
 
 /**
- * Check if a domain is blacklisted via IP ranges.
+ * Check if adresses are blacklisted via IP ranges.
  *
- * If it's not an IP already, resolve any addresses and check them all separately.
+ * @param {array} addresses             Adresses to check
+ * @returns {boolean}                   true if blacklisted
+ */
+function isBlacklisted(addresses) {
+    return Array.from(addresses).some(a => ipRangeCheck(a, ipRangeBlacklist));
+}
+
+/**
+ * Resolve a domain.
+ * 
+ * Throws an error if not resolvable.
  *
  * @param {string} domain           Domain to check
- * @returns {Promise<boolean>}      true if blacklisted
+ * @returns {array}                 The adresses resolved from the domain
+ * @throws                          On now resovlable domains
  */
-async function isDomainBlacklisted(domain) {
-    let addresses;
+ async function resolveDomain(domain) {
     if (!net.isIP(domain)) {
         try {
-            addresses = await dnsUtils.resolve(domain);
+            return await dnsUtils.resolve(domain);
         } catch (error) {
-            return true;
-        }
-        if (addresses.length === 0) {
-            return true;
+            throw Error('Hostname not resolvable.');
```

Comment:
This should be changed to return an empty list - we didn't raise before and we should avoid doing so. Or if we do, then the checks in `discoverHomeserverUrl` need to catch those errors - but it's cleaner here to just return an empty list, which is correct anyway since we resolved to no IP's for this domain.
```suggestion
            return [];
```